auth/src/main/java/com/firebase/ui/auth/ui/email/RegisterEmailFragment.java

```diff
@@ -210,6 +213,39 @@ public void onClick(View view) {
         });
     }
 
+    private void setUpPrivacyPolicy() {
+        if (TextUtils.isEmpty(mHelper.getFlowParams().privacyPolicyUrl)) {
+            return;
+        }
+
+        ForegroundColorSpan foregroundColorSpan =
+                new ForegroundColorSpan(ContextCompat.getColor(getContext(), R.color.linkColor));
+
+        String link = getString(R.string.privacy_policy);
+        SpannableStringBuilder spannableStringBuilder = new SpannableStringBuilder(link);
+        int start = mAgreementText.length();
+        spannableStringBuilder.setSpan(foregroundColorSpan, start, start + link.length(), 0);
+
+        mAgreementText.append(" and the ");
+        mAgreementText.append(spannableStringBuilder);
+        mAgreementText.setOnClickListener(new View.OnClickListener() {
```

Comment:
I needed a way to manually inject the strings so that I could get their start and end indices:
```java
    private void replaceTarget(String target, @StringRes int replacementRes, String url) {
        int targetIndex = mBuilder.toString().indexOf(target);
        if (targetIndex != -1) {
            String replacement = mContext.getString(replacementRes);
            mBuilder.replace(targetIndex, targetIndex + target.length(), replacement);

            int end = targetIndex + replacement.length();
            mBuilder.setSpan(mLinkSpan, targetIndex, end, 0);
            mBuilder.setSpan(new CustomTabsSpan(url), targetIndex, end, 0);
        }
    }
``` 

If you have any idea how to improve this mess, go for it! 😄